equinix/helpers_device.go

```diff
@@ -68,64 +69,46 @@ type NetworkInfo struct {
 	PrivateIPv4    string
 }
 
-func getNetworkInfoMetalGo(ips []metalv1.IPAssignment) NetworkInfo {
+func getNetworkInfo(ips []metalv1.IPAssignment) NetworkInfo {
 	ni := NetworkInfo{Networks: make([]map[string]interface{}, 0, 1)}
 	for _, ip := range ips {
 		network := map[string]interface{}{
-			"address": ip.Address,
-			"gateway": ip.Gateway,
-			"family":  ip.AddressFamily,
-			"cidr":    ip.Cidr,
-			"public":  ip.Public,
+			"address": ip.GetAddress(),
+			"gateway": ip.GetGateway(),
+			"family":  ip.GetAddressFamily(),
+			"cidr":    ip.GetCidr(),
+			"public":  ip.GetPublic(),
 		}
 		ni.Networks = append(ni.Networks, network)
 
 		// Initial device IPs are fixed and marked as "Management"
-		if *ip.Management {
-			if *ip.AddressFamily == 4 {
-				if *ip.Public {
-					ni.Host = *ip.Address
-					ni.IPv4SubnetSize = int(*ip.Cidr)
-					ni.PublicIPv4 = *ip.Address
+		if ip.GetManagement() {
+			if ip.GetAddressFamily() == 4 {
+				if ip.GetPublic() {
+					ni.Host = ip.GetAddress()
+					ni.IPv4SubnetSize = int(ip.GetCidr())
+					ni.PublicIPv4 = ip.GetAddress()
 				} else {
-					ni.PrivateIPv4 = *ip.Address
+					ni.PrivateIPv4 = ip.GetAddress()
 				}
 			} else {
-				ni.PublicIPv6 = *ip.Address
+				ni.PublicIPv6 = ip.GetAddress()
 			}
 		}
 	}
 	return ni
 }
 
-func getNetworkInfo(ips []*packngo.IPAddressAssignment) NetworkInfo {
-	ni := NetworkInfo{Networks: make([]map[string]interface{}, 0, 1)}
-	for _, ip := range ips {
-		network := map[string]interface{}{
-			"address": ip.Address,
-			"gateway": ip.Gateway,
-			"family":  ip.AddressFamily,
-			"cidr":    ip.CIDR,
-			"public":  ip.Public,
-		}
-		ni.Networks = append(ni.Networks, network)
-
-		// Initial device IPs are fixed and marked as "Management"
-		if ip.Management {
-			if ip.AddressFamily == 4 {
-				if ip.Public {
-					ni.Host = ip.Address
-					ni.IPv4SubnetSize = ip.CIDR
-					ni.PublicIPv4 = ip.Address
-				} else {
-					ni.PrivateIPv4 = ip.Address
-				}
-			} else {
-				ni.PublicIPv6 = ip.Address
-			}
+func getNetworkType(device *metalv1.Device) (*string, error) {
+	pgDevice := packngo.Device{}
```

Comment:
The `NetworkType` device attribute exposed by `packngo` is not an API-provided attribute; this serves as a shim so we can use the `packngo` logic to calculate `NetworkType`.  We could instead either duplicate the complete `packngo` logic here or copy it into `metal-go`, but my understanding is that `NetworkType` is not something we want to keep around anyway, so this shim is designed to get us through the SDK transition and then be eliminated when `packngo` is no longer used in the provider.